src/contracts/Value/ItemListInterface.php

```diff
@@ -29,4 +29,9 @@ public function has(string $identifier, string $language): bool;
      * @phpstan-param callable(\Ibexa\Contracts\Personalization\Value\ItemInterface=): bool $predicate
      */
     public function filter(callable $predicate): self;
+
+    /**
+     * Returns a new ItemInterface collection sliced of $length elements starting at position $offset.
+     */
+    public function slice(int $offset, ?int $length = null): self;
```

Comment:
Slices usually have a limit instead of a length. Any existing naming convention which led you to that?